modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SchemaHolderImpl.java

```diff
@@ -125,6 +128,8 @@ public AffinityIdentity(AffinityFunction aff, int backups, IgnitePredicate<Clust
     public SchemaHolderImpl(GridKernalContext ctx) {
         super(ctx);
 
+        this.ctx = ctx;
```

Comment:
KernalContext is always passed to the parent class,
Does it make sense to push 'ctx' field as well?

Reply:
Currently, only `SchemaHolderImpl` requires kernal context. We can move it to the parent class in the future when at least one more service will need it.